File: sdks/java/src/main/java/com/digows/whatsappgateway/messaging/SendMessageCommand.java
package com.digows.whatsappgateway.messaging;

import com.digows.whatsappgateway.command.OutboundCommand;
import com.digows.whatsappgateway.operational.SessionReference;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Outbound command that sends one WhatsApp message through one hosted session.
 * This remains the legacy/default outbound command on the shared NATS outgoing rail.
 */
public record SendMessageCommand(
  String commandId,
  SessionReference session,
  Message message
)
implements OutboundCommand
{
  @JsonProperty("action")
  public String action()
  {
    return "send";
  }
}

File: runtime/src/domain/entities/messaging/SendMessageCommand.ts
import { OutboundCommand, OutboundCommandFamily } from '../command/OutboundCommand.js';
import { SessionReference } from '../operational/SessionReference.js';
import { Message } from './Message.js';

/**
 * Application command that asks a session runtime to send one message.
 */
export class SendMessageCommand implements OutboundCommand {
  public readonly family = OutboundCommandFamily.Message;
  public readonly action = 'send';

  constructor(
    public readonly commandId: string,
    public readonly session: SessionReference,
    public readonly message: Message,
  ) {}
}
